quack/symmetric_gemm.py

```diff
@@ -0,0 +1,149 @@
+from typing import Optional
+from functools import partial
+from torch import Tensor
+from quack.gemm_act import GemmActMixin, GemmActSm90, act_fn_map, gemm_act
+from quack.gemm_sm90 import GemmSm90
+from quack.gemm_sm100 import GemmSm100
+from quack.tile_scheduler import TriangularTileScheduler
+from quack.gemm_wrapper_utils import GemmWrapperBase
+from quack.cute_dsl_utils import get_device_capacity, get_max_active_clusters
+import cutlass
+import cutlass.cute as cute
+import cutlass.torch as cutlass_torch
+from cutlass import Float32
+from cutlass.cute.runtime import make_ptr
+
+class GemmSymmetricMixin(GemmActMixin, GemmSm90):
+    def get_scheduler_class(self, varlen_m: bool = False):
+        return TriangularTileScheduler
+
+class GemmSymmetricSm90(GemmSymmetricMixin, GemmSm90):
+    pass
+
+class GemmSymmetricSm100(GemmSymmetricMixin, GemmSm100):
+    pass
+
+def gemm_symmetric(
+    A: Tensor,  # (l, m, k)
+    B: Tensor,  # (l, m, k)
+    D: Optional[Tensor],  # (l, m, m)
+    C: Optional[Tensor],  # (l, m, m) 
+    tile_count_semaphore: Optional[Tensor],  # (1,)
+    activation: Optional[str],
```

Comment:
I forgot, `activation` shouldn't be an argument here, it should be set to None inside the this function. Let's do it in a followup PR